pyproject.toml

```diff
@@ -0,0 +1,33 @@
+[build-system]
+requires = [
+    "setuptools>=61",
+    "setuptools-scm>=8.0"]
+build-backend = "setuptools.build_meta"
+
+[project]
+name = "fms-hf-tuning"
+version = "0.0.1"
+description = "FMS HF Tuning"
+authors = [
+  {name = "Sukriti Sharma", email = "sukriti.sharma4@ibm.com"},
+  {name = "Anh Uong", email = "anh.uong@ibm.com"},
+  {name = "Alex Brooks", email = "Alex.Brooks@ibm.com"},
+]
+license = {text = "Apache-2.0"}
+readme = "README.md"
+requires-python = "~=3.9"
+keywords = ['fms-hf-tuning', 'python', 'tuning']
+classifiers=[
+    "License :: OSI Approved :: Apache Software License"
+]
+
+dynamic = ["dependencies", "optional-dependencies"]
+
+[tool.setuptools.dynamic]
+dependencies = {file = ["requirements.txt"]}
+optional-dependencies = {flashattention = {file = ["flashattn_requirements.txt"]}}
```

Comment:
nit: since flashattn only has 1 dependency, instead of a separate file, should we use direct dependency name here? To avoid clutter of different requirement files ?

Reply:
nit: to keep the name of flashattention typed in same way everywhere, lets call the optional-dependency here as `flash-attn` as well, since this matches the original library name so atleast easier for people who are familiar with the library.

Reply:
Good suggestions @gkumbhat, I'm struggling a bit with the wheel pre-req, I think that @tedhtchang's point about it not belonging in the original requirements.txt file is valid, but when I move it back to the flashattn_requirements.txt file, it doesn't install properly.  I'll keep messing with it today, see if I can figure it out.